pkg/client/introduce/client.go

```diff
@@ -53,176 +38,63 @@ func New(ctx Provider, inv *didexchange.Invitation) (*Client, error) {
-}
+// WithPublicInvitation is used when introducer wants to provide public invitation.
+// NOTE: Introducer can provide invitation only after receiving RequestMsgType
+// USAGE: event.Continue(WithPublicInvitation(inv, to))
+var WithPublicInvitation = introduce.WithPublicInvitation // nolint: gochecknoglobals
 
-func (c *Client) handleOutbound(msg interface{}, o InvitationEnvelope) error {
-	payload, err := json.Marshal(msg)
-	if err != nil {
-		return fmt.Errorf("marshal outbound msg: %w", err)
-	}
-
-	didMsg, err := service.ParseDIDCommMsgMap(payload)
-	if err != nil {
-		return fmt.Errorf("new outbound DIDCommMsg msg: %w", err)
-	}
-
-	thID, err := didMsg.ThreadID()
-	if err != nil {
-		return fmt.Errorf("outbound threadID: %w", err)
-	}
-
-	if _, ok := msg.(*introduce.Request); !ok {
-		if err := c.saveInvitationEnvelope(thID, o); err != nil {
-			return err
-		}
-	}
-
-	return c.service.HandleOutbound(didMsg, o.Recps[0].MyDID, o.Recps[0].TheirDID)
-}
-
-// InvitationEnvelope keeps the information needed for sending a proposal
-type InvitationEnvelope struct {
-	// Invitation must be set when we have a public Invitation
-	Inv *didexchange.Invitation `json:"invitation,omitempty"`
-	// Recipients contain one or two elements
-	// one element - for the public Invitation, otherwise two elements
-	Recps []*introduce.Recipient `json:"recipients,omitempty"`
-}
-
-// Invitation returns an Invitation needed for the service (state machine depends on it)
-func (o *InvitationEnvelope) Invitation() *didexchange.Invitation {
-	return o.Inv
-}
-
-// Recipients returns data needed for the service (state machine depends on it)
-func (o *InvitationEnvelope) Recipients() []*introduce.Recipient {
-	return o.Recps
-}
-
-// sendProposal sends proposal
-func (c *Client) sendProposal(o InvitationEnvelope) error {
-	return c.handleOutbound(&introduce.Proposal{
-		Type: introduce.ProposalMsgType,
-		ID:   c.newUUID(),
-		To:   o.Recps[0].To,
-	}, o)
-}
-
-func (c *Client) saveInvitationEnvelope(thID string, o InvitationEnvelope) error {
-	data, err := json.Marshal(o)
-	if err != nil {
-		return fmt.Errorf("marshal invitation envelope: %w", err)
-	}
-
-	return c.store.Put(invitationEnvelopeKey(thID), data)
-}
-
-func (c *Client) getInvitationEnvelope(thID string) (*InvitationEnvelope, error) {
-	data, err := c.store.Get(invitationEnvelopeKey(thID))
-	if err != nil {
-		return nil, fmt.Errorf("get invitation envelope: %w", err)
-	}
-
-	var o *InvitationEnvelope
-	if err := json.Unmarshal(data, &o); err != nil {
-		return nil, fmt.Errorf("unmarshal invitation envelope: %w", err)
-	}
-
-	return o, nil
-}
-
-func invitationEnvelopeKey(thID string) string {
-	return invitationEnvelopePrefix + thID
-}
+// WithInvitation is used when introducee wants to provide invitation.
+// NOTE: Introducee can provide invitation only after receiving ProposalMsgType
+// USAGE: event.Continue(WithInvitation(inv))
+var WithInvitation = introduce.WithInvitation // nolint: gochecknoglobals
```

Comment:
why is there a global exported variable?

Reply:
It is a function alias. Instead of importing these functions from the service user can import them from the client.

Reply:
It is mutable - shouldn't have this.

Reply:
Done!